File: cpp/KendaraanDarat.cpp
#include "Kendaraan.cpp"
#include <iostream>
#include <string>

using namespace std;

class KendaraanDarat : public Kendaraan
{
private:
    int jumlahRoda;
    string jenisBahanBakar;
    string transmisi;

public:
    KendaraanDarat(
        string idKendaraan,
        string merk,
        int tahunProduksi,
        double harga,
        int jumlahRoda,
        string jenisBahanBakar,
        string transmisi) : Kendaraan(idKendaraan, merk, tahunProduksi, harga)
    {

        this->jumlahRoda = jumlahRoda;
        this->jenisBahanBakar = jenisBahanBakar;
        this->transmisi = transmisi;
    }

    bool cekKelayakanBerkendara()
    {
        return jumlahRoda >= 4 && getTahunProduksi() >= 2000;
    }

    void tampilkanInfoDarat()
    {
        tampilkanInfo();

        cout << "Jumlah Roda  : " << jumlahRoda << "\n";
        cout << "Bahan Bakar  : " << jenisBahanBakar << "\n";
        cout << "Transmisi    : " << transmisi << "\n";
    }

    int getJumlahRoda()
    {
        return jumlahRoda;
    }

    string getJenisBahanBakar()
    {
        return jenisBahanBakar;
    }

    string getTransmisi()
    {
        return transmisi;
    }
};

File: java/KendaraanDarat.java
public class KendaraanDarat extends Kendaraan {
  private int jumlahRoda;
  private String jenisBahanBakar;
  private String transmisi;

  public KendaraanDarat(
      String idKendaraan,
      String merk,
      int tahunProduksi,
      double harga,
      int jumlahRoda,
      String jenisBahanBakar,
      String transmisi) {

    super(idKendaraan, merk, tahunProduksi, harga);

    this.jumlahRoda = jumlahRoda;
    this.jenisBahanBakar = jenisBahanBakar;
    this.transmisi = transmisi;
  }

  public boolean cekKelayakanBerkendara() {
    return jumlahRoda >= 4 && getTahunProduksi() >= 2000;
  }

  public void tampilkanInfo() {
    super.tampilkanInfo();

    System.out.println("Jumlah Roda  : " + jumlahRoda);
    System.out.println("Bahan Bakar  : " + jenisBahanBakar);
    System.out.println("Transmisi    : " + transmisi);
  }

  public int getJumlahRoda() {
    return jumlahRoda;
  }

  public String getJenisBahanBakar() {
    return jenisBahanBakar;
  }

  public String getTransmisi() {
    return transmisi;
  }
}